collector/powersupplyclass.go

```diff
@@ -55,7 +56,7 @@ func NewPowerSupplyClassCollector(logger log.Logger) (Collector, error) {
 func (c *powerSupplyClassCollector) Update(ch chan<- prometheus.Metric) error {
 	powerSupplyClass, err := getPowerSupplyClassInfo(c.ignoredPattern)
 	if err != nil {
-		if os.IsNotExist(err) {
+		if errors.Is(err, os.ErrNotExist) {
```

Comment:
```suggestion
		if os.IsNotExist(err) {
```

Reply:
This is needed for syscalls

Reply:
It seems to work just fine with the ReadDir result.